recipes/aeron/all/conanfile.py

```diff
@@ -0,0 +1,142 @@
+        "fPIC": True,
+        "build_aeron_driver": True,
+        "build_aeron_archive_api": True
+    }
+    generators = "cmake"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+        compiler = str(self.settings.compiler)
+        compiler_version = tools.Version(self.settings.compiler.version)
+
+        minimal_version = {
+            "Visual Studio": "16",
+            "gcc": "5"
+        }
+
+        if compiler in minimal_version and compiler_version < minimal_version[compiler]:
+            raise ConanInvalidConfiguration(
+                "{} requires {} compiler {} or newer [is: {}]".format(self.name, compiler, minimal_version[compiler], compiler_version)
+            )
+
+    def build_requirements(self):
+        self.build_requires("zulu-openjdk/11.0.8")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 11)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["AERON_INSTALL_TARGETS"] = True
+        self._cmake.definitions["BUILD_AERON_DRIVER"] = self.options.build_aeron_driver
+        self._cmake.definitions["AERON_TESTS"] = False
+        self._cmake.definitions["AERON_BUILD_SAMPLES"] = False
+        self._cmake.definitions["BUILD_AERON_ARCHIVE_API"] = self.options.build_aeron_archive_api
+        self._cmake.definitions["AERON_ENABLE_NONSTANDARD_OPTIMIZATIONS"] = True
+
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _patch_sources(self):
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"), "/MTd", "")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"), "/MT", "")
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        with tools.chdir(self.package_folder):
+            tools.rename("include", "old_include")
+            os.makedirs("include")
+            tools.rename("old_include", os.path.join("include", "aeron"))
+
+            for dll in glob.glob(os.path.join("lib", "*.dll")):
+                shutil.move(dll, "bin")
```

Comment:
litte confused about 😕 https://github.com/conan-io/conan-center-index/pull/5035/commits/c2f1a7ee35ebd58d861f9c29caa9b342cd6409cf and https://github.com/lukaszlaszko/conan-center-index/pull/1/files#diff-cd2c794b65ea18edbc2969d0414a9943932959a28c71a6955e5b6ff85eeae02fL110

As per upstream it supposed to be in `include` so I am not sure this is required

https://github.com/real-logic/aeron/blob/e5d15bee16f4aee1e585daa7b6ea9f39391e6889/aeron-samples/src/main/cpp/Ping.cpp#L31


```suggestion
        for dll in glob.glob(os.path.join("lib", "*.dll")):
            shutil.move(dll, "bin")
```

Reply:
@prince-chrismc The idea was to standardize towards approach of different libraries. But I agree it probably wasn't the intention of the library authors. Let's follow the upstream indeed. I've made the changes. Thanks